plugins/lookup/vault_login.py

```diff
@@ -0,0 +1,124 @@
+# (c) 2021, Brian Scholer (@briantist)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+  lookup: vault_login
+  version_added: 2.2.0
+  author:
+    - Brian Scholer (@briantist)
+  short_description: Perform a login operation against HashiCorp Vault
+  requirements:
+    - C(hvac) (L(Python library,https://hvac.readthedocs.io/en/stable/overview.html))
+    - For detailed requirements, see R(the collection requirements page,ansible_collections.community.hashi_vault.docsite.user_guide.requirements).
+  description:
+    - Performs a login operation against a given path in HashiCorp Vault, returning the login response, including the token.
+  seealso:
+    - module: community.hashi_vault.vault_login
+    - ref: community.hashi_vault.vault_login_token filter <ansible_collections.community.hashi_vault.docsite.filter_guide.vault_login_token>
+      description: The official documentation for the C(community.hashi_vault.vault_login_token) filter plugin.
+  notes:
+    - This lookup does not use the term string and will not work correctly in loops. Only a single response will be returned.
+    - "A login is a write operation (creating a token persisted to storage), so this module always reports C(changed=True),
+      except when used with C(token) auth, because no new token is created in that case. For the purposes of Ansible playbooks however,
+      it may be more useful to set C(changed_when=false) if you're doing idempotency checks against the target system."
+    - The C(none) auth method is not valid for this plugin because there is no response to return.
+    - "With C(token) auth, no actual login is performed.
+      Instead, the given token's additional information is returned in a structure that resembles what login responses look like."
+    - "The C(token) auth method will only return full information if I(token_validate=True).
+      If the token does not have the C(lookup-self) capability, this will fail. If I(token_validate=False), only the token value itself
+      will be returned in the structure."
+  extends_documentation_fragment:
+    - community.hashi_vault.connection
+    - community.hashi_vault.connection.plugins
+    - community.hashi_vault.auth
+    - community.hashi_vault.auth.plugins
+  options:
+    _terms:
+      description: This is unused and should be an empty string.
+      type: str
+      required: True
+"""
+
+EXAMPLES = """
+- name: Set a fact with a lookup result
+  set_fact:
+    login_data: "{{ lookup('community.hashi_vault.vault_login', '', url='https://vault', auth_method='userpass', username=user, password=pwd) }}"
```

Comment:
Like this:
```suggestion
    login_data: "{{ lookup('community.hashi_vault.vault_login', url='https://vault', auth_method='userpass', username=user, password=pwd) }}"
```

Reply:
Thank you, confirmed this works and updated in https://github.com/ansible-collections/community.hashi_vault/pull/199/commits/c95599bf48ed30b3f264ee6c4973e27abb4c14f3